notebooks/kaggle/store-sales-time-series-forecasting/Time_Series_Forecasting.ipynb

```diff
@@ -0,0 +1,1515 @@
+    "<i>As we can see from these characteristics, more than 25% of observations have 0 in 'sales' column. Maybe this can be influenced by some categoies, which mostly have zero number of sales. It is an interesting point to explore. <br>What is also interesting about 'sales' column is that 87.5% of observations have less than 640 number of sales, but <b>max</b> value is much bigger. Maybe it is connected with some very popular categories, maybe it is a consequence of some event (I can only guess now, but it can be connected with the earthquake in Ecuador, which happened in Ecuador in 2016. If it is true, such observations could be classified as outliers, because they won't help us with building a predictive model).</i>\n",
+    "![image.png](attachment:image.png) (screenshot from the task description in Kaggle)\n",
+    "<p><i>The same situation with the 'transactions' feature. There is a much bigger <b>max</b> number of transactions in comparison with 87.5% of 'transactions' values. The reason can be the same as I mentioned before, but it is necessary to check it out.</i>\n",
+    "<p><i>One more interesting fact is that the number of promotions is very small in comparison with the whole number of observations (less than 25%). But there is no sense to delete this feature right now, because it seems, that promotions have a good impact on the target.</i>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Numerical data distribution"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(20, 40))\n",
+    "for i, current_num_feature in enumerate(numerical_features):\n",
+    "    plt.subplot(len(numerical_features), 1, i + 1)\n",
+    "    plt.xlabel(current_num_feature)\n",
+    "    plt.ylabel('count')\n",
+    "    x = train_data.groupby('date')[current_num_feature].mean()\n",
+    "    plt.hist(x=x, bins=80)\n",
+    "    plt.title(current_num_feature)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<i>I'll describe these 4 plots one by one:<br> 1) 'Sales' plot. Distribution of means is not standard, and it is hard to say something about it, having no connection with datetime. But it interesting to explore values, which are out of the general distribution.</i>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "date_and_mean_sales = train_data.groupby('date')['sales'].mean()\n",
+    "date_and_mean_sales_min = date_and_mean_sales[date_and_mean_sales < 100]\n",
+    "date_and_mean_sales_min"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<i>It means, that all the values in the left side of 'sales' distribution refer to New Year. That's why we can easily remove these rows - such data are <b>OUTLIERS</b></i>."
```

Comment:
In time series problems outlier values should not be necessarily removed (they can be removed, but this leads to gaps in the time series), they can also be smoothened (with rolling averages) or decreased in their importance with low sample weights.